ucs/domain/interfaces/inbound/upload_service.py

```diff
@@ -0,0 +1,199 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Interfaces for the main upload handling logic of this service."""
+
+from typing import Protocol
+
+from ucs.domain import models
+
+# shortcuts:
+# pylint: disable=unused-import
+from ucs.domain.interfaces.inbound.file_service import FileUnkownError  # noqa: F401
+
+
+class UploadUnkownError(RuntimeError):
+    """Thrown when an upload attempt with the given ID is not known."""
+
+    def __init__(self, *, upload_id: str):
+        self.upload_id = upload_id
+        message = f"The upload attempt with ID {upload_id} is unkown."
+        super().__init__(message)
+
+
+class UploadStatusMissmatchError(RuntimeError):
+    """Thrown when an upload was expected to have a specific status for performing
+    an action."""
+
+    def __init__(
+        self,
+        *,
+        upload_id: str,
+        expected_status: models.UploadStatus,
+        current_status: models.UploadStatus,
+    ):
+        self.upload_id = upload_id
+        self.expected_status = expected_status
+        self.current_status = current_status
+        message = (
+            f"The upload with ID {upload_id} must be in '{expected_status}' state to"
+            + " perform the requested action, however, its current state is:"
+            + str(current_status)
+        )
+        super().__init__(message)
+
+
+class ExistingActiveUploadError(RuntimeError):
+    """Thrown when trying to create a new upload while there is another upload active."""
+
+    def __init__(self, *, active_upload: models.UploadAttempt):
+        self.active_upload = active_upload
+        message = (
+            "Failed to create a new multi-part upload for the file with ID"
+            + f" {self.active_upload.file_id} because another upload is"
+            + " currently active or has been accepted."
+            + f" ID, status of the existing upload: {self.active_upload.upload_id},"
+            + f" {self.active_upload.status}"
+        )
+        super().__init__(message)
+
+
+class UploadCompletionError(RuntimeError):
+    """Thrown when the confirmation of an upload attempt failed."""
+
+    def __init__(self, *, upload_id: str, reason: str):
+        self.reason = reason
+        self.upload_id = upload_id
+        message = (
+            f"The confirmation of the upload attempt with ID {upload_id} failed."
+            + " The upload attempt was aborted and cannot be resumed. The reason was:"
+            + reason
+        )
+        super().__init__(message)
+
+
+class UploadCancelError(RuntimeError):
+    """Thrown when the cancelling of an upload attempt failed."""
+
+    def __init__(self, *, upload_id: str):
+        self.upload_id = upload_id
+        self.possible_reason = (
+            "An ongoing part upload might be a reason. Please complete all part upload"
```

Comment:
```suggestion
            "An ongoing part upload might be a reason. Please complete all part uploads"
```

Reply:
This seems weird... shouldn't we take care of cancelling the multipart uploads in this case?